### PR TITLE
Bug fix/internal issue #725

### PR DESCRIPTION
**Pull request by Dronplane**
Analyzers revision reading is reworked to include not only current but also system database revision in query json.
To reduce json size MIN revision is default and never  included in json so if current and/or system database contains no custom analyzers -> nothing is transferred.

Also to keep backward compatibility revision value was excluded from analyzer properties output for user.

### CODE REVIEW

File: arangod/Cluster/ClusterTypes.cpp [hunk 1]
```diff
@@ -25,13 +25,92 @@
...
+    scope->add(StaticStrings::ArangoSearchCurrentAnalyzersRevision,
+               VPackValue(currentDbRevision));
+  }
+  if (systemDbRevision != AnalyzersRevision::MIN) {
+    scope->add(StaticStrings::ArangoSearchSystemAnalyzersRevision,
+               VPackValue(systemDbRevision));
+  }
+}
+
+bool AnalyzersRevision::QueryAnalyzerRevisions::fromVelocyPack(velocypack::Slice slice, std::string& error) {
```
Comment (gnusi) [inline arangod/Cluster/ClusterTypes.cpp:52]: return `Result` instead?

File: arangod/Cluster/ClusterTypes.h [hunk 2]
```diff
@@ -86,9 +86,51 @@ struct AnalyzersRevision {
...
+
   static constexpr Revision LATEST = std::numeric_limits<uint64_t>::max();
   static constexpr Revision MIN = 0;
 
+  /// @brief Analyzers revisions used in query.
+  /// Stores current database revision
+  /// and _system database revision (analyzers from _system are accessible from other databases)
+  /// If at some point we will decide to allow cross-database anayzer usage this could
+  /// became more complicated. But for now  we keep it simple - store just two members
+  struct QueryAnalyzerRevisions {
```
Comment (gnusi) [inline arangod/Cluster/ClusterTypes.h:98]: let's move it out of `AnalyzersRevision` scope

File: arangod/Cluster/ClusterTypes.h [hunk 3]
```diff
@@ -86,9 +86,51 @@ struct AnalyzersRevision {
...
+      return !(*this == other);
+    }
+
+    /// @brief Gets analyzers revision to be used with specified database
+    /// @param vocbase database name
+    /// @return analyzers revision
+    Revision getVocbaseRevision(DatabaseID const& vocbase) const noexcept;
+
+    Revision currentDbRevision{ AnalyzersRevision::MIN};
+    Revision systemDbRevision{ AnalyzersRevision::MIN};
```
Comment (gnusi) [inline arangod/Cluster/ClusterTypes.h:129]: I suggest to mark both members with `const` to emphasize immutability

Comment (Dronplane) [reply arangod/Cluster/ClusterTypes.h]: Good point about enforcing immutability. But let`s just make members private (as actually no direct access is needed) and give no setters. Making members const will complicate fromVelocyPack implementation.

**Comment by Dronplane** [comment]
https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/10677/

**Comment by Dronplane** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/10687/ Tests blue

**Pull request merged**